### PR TITLE
PLAT-608 Ensure deterministic row order during burst validation

### PR DESCRIPTION
**Pull request by alexander-schmidt**

### CODE REVIEW

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ public DbTableRowNotificationSet(IDbTable<R, ?> table) {
 	}
 
 	@Override
-	public Collection<DbTableRowNotification<R>> getAllNotifications() {
+	public List<DbTableRowNotification<R>> getAllNotifications() {
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java:25]: why change this?

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java [hunk 2]
```diff
@@ -14,36 +14,36 @@
  */
 public interface IDbTableRowNotificationSet<R> {
 
-	Collection<DbTableRowNotification<R>> getAllNotifications();
+	List<DbTableRowNotification<R>> getAllNotifications();
 
-	default Set<R> getAllTableRows() {
+	default List<R> getAllTableRows() {
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java:19]: suggestion: return `Collection` here and everywhere below

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
 
 	private final IDbTable<R, ?> table;
-	private final Set<DbTableRowNotification<R>> notifications;
+	private final List<DbTableRowNotification<R>> notifications;
 
 	public DbTableRowNotificationSet(IDbTable<R, ?> table) {
 
 		this.table = table;
-		this.notifications = new HashSet<>();
+		this.notifications = new HashList<>();
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java:16]: the set was preventing a duplication, if we change it to `HashList`, this is no longer the case

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java [hunk 4]
```diff
@@ -14,36 +14,36 @@
...
 
 		return getAllNotifications()//
 			.stream()
 			.map(DbTableRowNotification::getTableRow)
-			.collect(Collectors.toCollection(HashSet::new));
+			.collect(Collectors.toCollection(HashList::new));
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java:24]: here again, we remove the de-duplication when going to `List`

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java (hunk 3 above)
Comment (alexander-schmidt) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java:16]: Yes it is. That's the whole point of `HashList`. See `HashList.add`:
```
	@Override
	public boolean add(T element) {

		if (set.add(element)) {
			list.add(element);
			return true;
		} else {
			return false;
		}
	}
```

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java:25]: To imply that elements are contained in a specific order.

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java (hunk 3 above)
Comment (oliver-richers) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java:16]: you are right, I forgot about that :)

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/DbTableRowNotificationSet.java]: (this change was reverted)

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java (hunk 4 above)
Comment (alexander-schmidt) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/listener/IDbTableRowNotificationSet.java:24]: not correct; see above

**Review by oliver-richers — APPROVED** [review]
fine, but we should add some tests

**Pull request merged**